src/Wrappers.dfy

```diff
@@ -69,4 +69,26 @@ module Wrappers {
       value
     }
   }
+
+  datatype MUST<E> = NeedsAreMeet | ResultInError(error: E)
```

Comment:
I was thinking about the dual of option. I think that better reflects what is going on. I see your "But wait, I can get a `Result`" comment, but I would argue that this is the same as `Option`. Because "when we don't need to pass around a reason for the failure," is how an option is described. To me this means that `Result` is the "complete" type and `Option` and `Outcome` are the "lossy" forms. It is just that we tend to prioritize the value so it may feel weird(?)

In the case of `Need` we care about the error, and the transformation into a `Result` is because this "focus on the error" is the same kind of syntactic sugar we are looking at with `Option`. That my pitch anyway :)